scripts/rollup/modules.js

```diff
@@ -24,6 +24,7 @@ const importSideEffects = Object.freeze({
 const knownGlobals = Object.freeze({
   react: 'React',
   'react-dom': 'ReactDOM',
+  'react-scheduler': 'ReactScheduler',
```

Comment:
I don’t think we want this. We *don’t* want it to be used in UMD builds if I understand it right.

Reply:
Instead we want to add `react-scheduler` to `dependencies` of `react-dom`.

Reply:
What does `dependencies` mean in this context? Things that get inlined into the flat bundle, or things that are left as a require statement (or global reference, for UMD builds)?

The point of the scheduler package is to coordinate React work with non-React work (e.g. Booloader, Webpack, non-React product code) using a centralized scheduler. We want the scheduler to be packaged separately from React and React DOM, not inlined into the flat bundle, so everything can share the same instance.

Since this is a breaking change for UMD bundles, the plan for 16.x is to include the scheduler in the UMD bundle but use the global scheduler if it's available. In 17+, we stop bundling and always use the global scheduler instance.

For CommonJS bundles, it's not a breaking change, so we can externalize it immediately without waiting for 17.

Reply:
AFAIK putting it in dependencies should do exactly what you described (for 16).

CJS gets a require, UMD continues bundling (if you remove it from knownGlobals).